<?xml version="1.0" encoding="UTF-8"?>

<launch>

    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="5" />
    <arg name="roll" default="0"/>
    <arg name="pitch" default="3.14159"/>
    <arg name="yaw" default="0" />

    <arg name="sdf_robot_file" default="$(find optimal_visual_servoing)/data/urdf/aruco.sdf" />
    <arg name="robot_name" default="aruco" />

    <include file="$(find optimal_visual_servoing)/launch/spawn_robot_sdf.launch">
        <arg name="x" value="$(arg x)" />
        <arg name="y" value="$(arg y)" />
        <arg name="z" value="$(arg z)" />
        <arg name="roll" value="$(arg roll)"/>
        <arg name="pitch" value="$(arg pitch)"/>
        <arg name="yaw" value="$(arg yaw)" />

        <arg name="sdf_robot_file" value="$(arg sdf_robot_file)" />
        <arg name="robot_name" value="$(arg robot_name)" />
    </include>

    <node name="tag_trajectory_generator" pkg="optimal_visual_servoing" type="tag_trajectory_generator" respawn="false" output="screen"/>

    <!-- <node name="optimal_visual_servoing" pkg="optimal_visual_servoing" type="optimal_visual_servoing" respawn="false" output="screen" /> -->
</launch>
